package/setup.py

```diff
@@ -17,37 +16,6 @@
 
 files = map(lambda x: x.replace("kedro_viz/", "", 1), jsbuild)
 
-with open(path.join(here, path.pardir, "package.json")) as data:
-    obj = json.load(data)
-    version = obj["version"]
-
-with open("requirements.txt", "r", encoding="utf-8") as f:
-    requires = [x.strip() for x in f if x.strip()]
-
-with open(path.join(here, "../README.md"), encoding="utf-8") as f:
-    readme = f.read()
-
-
 setup(
-    name=name,
-    version=version,
-    description="Kedro-Viz helps visualise Kedro data and analytics pipelines",
-    long_description=readme,
-    long_description_content_type="text/markdown",
-    license="Apache Software License (Apache 2.0)",
-    url="https://github.com/kedro-org/kedro-viz",
-    python_requires=">=3.7",
-    install_requires=requires,
-    keywords="pipelines, machine learning, data pipelines, data science, data engineering, visualisation",
-    author="Kedro",
-    packages=find_packages(exclude=["tests*", "features*"]),
     package_data={"kedro_viz": list(files)},
```

Comment:
@MehdiNV I tested removing the package_data and also deleted the setup.py, setup.cfg completely. I was able to install editable package via `pip install -e package` and also was able to run `kedro viz`. I am not sure where the jsbuild is used tbh. 

@astrojuanlu - is the jsbuild used during publishing the package ? We execute the command `make package` and the below script runs -
```makefile
package:
	find . -regex ".*/__pycache__" -exec rm -rf {} +
	find . -regex ".*\.egg-info" -exec rm -rf {} +
	test -f package/kedro_viz/html/index.html || (echo "Built npm package not found; packaging process cancelled."; exit 1)
	cd package && python setup.py clean --all
	cd package && python setup.py sdist bdist_wheel
``` 
@MehdiNV we might need to change these steps as well if we get rid of setup.py completely

@astrojuanlu - how can we test if the migration is success, apart from installing the package locally ? We also need to test the `make package` command works and the package includes everything as before ?

Thank you

Reply:
@ravi-kumar-pilla I think setup.py is only needed as a fallback for users who cannot utilise pyproject.toml for one reason or another ([see stackoverflow post on it](https://stackoverflow.com/questions/73310069/should-i-be-using-only-pyproject-toml)), what do you think about keeping it as a plan B? Or perhaps it's not necessary as it's very unlikely a user will have incompatability w/ .toml?

Reply:
I think we can migrate the requirements.txt, test_requirements.txt, setup.cfg and part of setup.py (excluding the readme and package data) to pyproject.toml. I want to discover more on this and see how can we test the setup locally and how it affects our release process of `make package ` I mentioned above. Will work on this today !

Reply:
I think the way to test if the migration is successful is producing the final `.tar.gz` and `.whl` files that go to https://pypi.org/project/kedro-viz/#files, with and without this PR, and making sure that the relevant files are identical

Reply:
Hi @astrojuanlu , 

Thank you for the input. I tested them and they seem to be the same. I have a question on installing dependencies. 

Previously when we used to install dependencies of kedro-viz via requirements.txt file, kedro-viz was not installed as a package. However, when installing the same dependencies via pyproject.toml, the list of installed dependencies contain kedro-viz. 

I checked this for kedro-datasets and even kedro-datasets get installed as dependency when we install its dependencies from pyproject.toml. Is this normal ?

Thank you

Reply:
Yes, because there is no such a thing as `pip install . --only-deps` https://github.com/pypa/pip/issues/11440

If you want to still be able to install only the dependencies but _not_ kedro-viz, you should probably keep `requirements.txt` around and read it from `pyproject.toml`, declaring it as dynamic https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#dynamic-metadata 

```toml
dependencies = {file = "requirements.text"}}
```

(didn't test this syntax)

Reply:
It's actually `dependencies = {file = "requirements.txt"}`

Reply:
Hi @astrojuanlu , Thank you for the suggestion. 

I have a question regarding keeping the requirements file as is. Was the whole idea was to move away from the requirements file ? What is the advantage of using pyproject.toml and then loading the deps from requirements file. Thank you

Reply:
This:

> If you want to still be able to install only the dependencies but not kedro-viz

Reply:
Caveat: https://github.com/kedro-org/kedro-viz/issues/1582#issuecomment-1764451560

Reply:
Since we are not moving to pyproject.toml completely, I do not see why this migration is needed. For handling the caveat mentioned, I feel we should go by the option - `Configure dependabot to use widen.` and migrate to pyproject.toml when the ticket [pypa/pip#11440](https://github.com/pypa/pip/issues/11440) is merged. What do you think of this approach ? @astrojuanlu @tynandebold

Reply:
I'm supportive of that 👍🏽

Reply:
^ This makes sense; the .toml file is now linked to a `requirements.txt` file containing the dependencies, however I've kept the `test_requirements.txt` file deleted (installing test dependencies is via .toml). 

In my head, a user should have no need for the test-dependencies unless they need to use / install kedro-viz as well (in which case, they'd use the .toml fully anyway). Is this a suitable approach?

Reply:
@MehdiNV we will not go ahead with this draft PR. Instead, we will use the option of `Configuring dependabot to use widen.` to handle - https://github.com/kedro-org/kedro-viz/issues/1582#issuecomment-1764451560

Reply:
Ahhh makes sense; apologies, I hadn't reloaded the page before so hadn't seen the new comments in the last hour when I was replying. Will shift towards the widen option then - will close this PR in the meantime as it's redundant